hi-ml-histopathology/src/histopathology/utils/output_utils.py

```diff
@@ -147,29 +143,30 @@ def save_features(results: ResultsType, outputs_dir: Path) -> None:
     torch.save(features_list, outputs_dir / 'test_encoded_features.pickle')
 
 
-def save_top_and_bottom_tiles(results: ResultsType, n_classes: int, figures_dir: Path) \
+def save_top_and_bottom_tiles(results: ResultsType, n_classes: int, figures_dir: Path, k_tiles: int = 10) \
         -> Dict[str, List[str]]:
     logging.info("Selecting tiles ...")
 
-    def select_k_tiles_from_results(label: int, select: Tuple[str, str]) \
+    def select_k_tiles_from_results(label: int, highest_pred: bool, highest_att: bool) \
             -> List[Tuple[Any, Any, List, List]]:
-        return select_k_tiles(results, n_slides=10, label=label, n_tiles=10, select=select)
+        return select_k_tiles(results, n_slides=k_tiles, label=label, n_tiles=k_tiles, highest_pred=highest_pred,
+                              highest_att=highest_att)
 
     # Class 0
-    tn_top_tiles = select_k_tiles_from_results(label=0, select=('highest_pred', 'highest_att'))
-    tn_bottom_tiles = select_k_tiles_from_results(label=0, select=('highest_pred', 'lowest_att'))
-    fp_top_tiles = select_k_tiles_from_results(label=0, select=('lowest_pred', 'highest_att'))
-    fp_bottom_tiles = select_k_tiles_from_results(label=0, select=('lowest_pred', 'lowest_att'))
+    tn_top_tiles = select_k_tiles_from_results(label=0, highest_pred=True, highest_att=True)
+    tn_bottom_tiles = select_k_tiles_from_results(label=0, highest_pred=True, highest_att=False)
+    fp_top_tiles = select_k_tiles_from_results(label=0, highest_pred=False, highest_att=True)
+    fp_bottom_tiles = select_k_tiles_from_results(label=0, highest_pred=False, highest_att=False)
     report_cases = {'TN': [tn_top_tiles, tn_bottom_tiles],
```

Comment:
do those strings "TN"/"FP"... need to be the same as in the output handler? If so, apply DRY

Reply:
deprecated